pkg/apis/core/v1beta1/helper/shoot.go

```diff
@@ -691,3 +692,40 @@ func ControlPlaneNamespaceForShoot(shoot *gardencorev1beta1.Shoot) string {
 	}
 	return shoot.Status.TechnicalID
 }
+
+// IsUpdateStrategyInPlace returns true if the given machine update strategy is either AutoInPlaceUpdate or ManualInPlaceUpdate.
+func IsUpdateStrategyInPlace(updateStrategy *gardencorev1beta1.MachineUpdateStrategy) bool {
+	if updateStrategy == nil {
+		return false
+	}
+
+	return *updateStrategy == gardencorev1beta1.AutoInPlaceUpdate || *updateStrategy == gardencorev1beta1.ManualInPlaceUpdate
+}
+
+// CalculateDataStringForKubeletConfiguration returns a data string for the relevant fields of the kubelet configuration.
+func CalculateDataStringForKubeletConfiguration(kubeletConfiguration *gardencorev1beta1.KubeletConfig) []string {
+	data := []string{}
+
+	if kubeletConfiguration == nil {
+		return data
+	}
+
+	if resources := SumResourceReservations(kubeletConfiguration.KubeReserved, kubeletConfiguration.SystemReserved); resources != nil {
+		data = append(data, fmt.Sprintf("%s-%s-%s-%s", resources.CPU, resources.Memory, resources.PID, resources.EphemeralStorage))
+	}
+	if eviction := kubeletConfiguration.EvictionHard; eviction != nil {
+		data = append(data, fmt.Sprintf("%s-%s-%s-%s-%s",
+			ptr.Deref(eviction.ImageFSAvailable, ""),
+			ptr.Deref(eviction.ImageFSInodesFree, ""),
+			ptr.Deref(eviction.MemoryAvailable, ""),
+			ptr.Deref(eviction.NodeFSAvailable, ""),
+			ptr.Deref(eviction.NodeFSInodesFree, ""),
+		))
+	}
+
+	if policy := kubeletConfiguration.CPUManagerPolicy; policy != nil {
+		data = append(data, *policy)
+	}
+
+	return data
+}
```

Comment:
I think this better belongs to `pkg/utils/gardener` or `pkg/utils/kubernetes`, WDYT?